Error reporting for add_depfileset

### Feature Description

It happens that I forget to add parenthes. ex below:

  with self.active_fileset('rtl'):
            self.add_depfileset(MemAgent)

The error message says the type is wrong, since this may be quite a common mistake it would be nice to report back that a class and not an object was found (ie. type(MemAgent) vs type(MemAgent())


           self.add_dep(dep, clobber=True)
        else:
>           raise TypeError("dep is not a valid type")
E           TypeError: dep is not a valid type

/home/aolofsson/work/zeroasic/siliconcompiler/siliconcompiler/design.py:371: TypeError

--- a/siliconcompiler/design.py
+++ b/siliconcompiler/design.py
@@ -368,7 +368,7 @@ class Design(LibrarySchema, DependencySchema):
             if dep is not self:
                 self.add_dep(dep, clobber=True)
         else:
-            raise TypeError(f"dep is not a valid type: {dep} ({type(dep)})")
+            raise TypeError(f"dep is not a valid type: {dep}")
 
         if not isinstance(dep, Design):
             raise ValueError(f"cannot associate fileset ({depfileset}) with {dep.name}")

--- a/tests/test_design.py
+++ b/tests/test_design.py
@@ -361,14 +361,13 @@ def test_options_depfileset_with_selfname():
 
 def test_options_depfileset_with_invalid_input():
     with pytest.raises(TypeError,
-                       match=r"dep is not a valid type: 1 \(<class 'int'>\)"):
+                       match="dep is not a valid type: 1"):
         Design("test").add_depfileset(1, "rtl", "rtl")
 
 
 def test_options_depfileset_with_invalid_cls():
     with pytest.raises(TypeError,
-                       match="dep is not a valid type: <class 'siliconcompiler.design.Design'> "
-                             r"\(<class 'type'>\)"):
+                       match="dep is not a valid type: <class 'siliconcompiler.design.Design'>"):
         Design("test").add_depfileset(Design, "rtl", "rtl")
 
 

